lib/AST/Module.cpp

```diff
@@ -2574,6 +2574,25 @@ bool SourceFile::isImportedAsSPI(const ValueDecl *targetDecl) const {
   return false;
 }
 
+bool SourceFile::importsModuleAsWeakLinked(const ModuleDecl *module) const {
+  for (auto &import : *Imports) {
+    if (!import.options.contains(ImportFlags::WeakLinked))
+      continue;
+
+    const ModuleDecl *importedModule = import.module.importedModule;
+    if (module == importedModule)
+      return true;
+
+    // Also check whether the target module is actually the underlyingClang
+    // module for this @_weakLinked import.
```

Comment:
Can we add a test case for the Swift overlay + underlying Clang module case?

Reply:
This is tested by the `weaklinked_import_clang` test case (I discovered needing this logic while writing it).

Reply:
I see. Thanks!